src/path.rs

```diff
@@ -21,12 +20,16 @@ impl fmt::Display for Fragment {
     }
 }
 
-#[derive(Clone, Debug, Deserialize)]
+#[derive(Clone, Debug, Deserialize, PartialEq, Eq)]
+#[serde(try_from = "&str")]
 pub struct Path {
-    fragments: Vec<Fragment>,
-    is_relative: bool,
+    pub fragments: Vec<Fragment>,
+    pub is_relative: bool,
 }
 
+#[derive(Debug)]
+pub struct PathError(&'static str);
```

Comment:
This can be addressed in another PR, but it's reminded me to look at the other error types and it would be worthwhile to refactor our error type to use thiserror.

When we do that we can decide whether this should be a part of some broader common error

Reply:
Refactoring error types would be valuable. `PathError` and `ContainerError` were only created for type-tetris reasons, and could be a lot cleaner. The original error code in `error.rs` uses methods of `error::Error` that have since been deprecated.

Reply:
I see the issue for it at https://github.com/juniperparsnips/rink-runtime/issues/9.